### DISCUSSION TITLE
Why can't my wallet sync?

### DISCUSSION THREAD

**Discussion by opencoin-op** [Chia-Network/chia-blockchain#16928] [Support]
run.sh
------------------------

```
#!/bin/bash

VERSION=2.1.1

docker build --build-arg VERSION=$VERSION -t xch:$VERSION .
docker stop xch
docker rm xch
docker run --init -d --restart=always --name xch -m 96G -p 8555:8555 -p 9256:9256 -v /data2/xch:/root/.chia xch:$VERSION
```

Dockerfile
------------------------------
```
FROM ubuntu:20.04

ARG VERSION=2.1.1

# install dependencies
RUN apt-get update \
    && apt-get upgrade -y \
    && apt-get install python3 sudo net-tools git curl -y \
    && DEBIAN_FRONTEND=noninteractive apt-get install -y lsb-core

# install chia
RUN cd $HOME \
    && git clone https://github.com/Chia-Network/chia-blockchain.git -b latest \
    && cd chia-blockchain \
    && git checkout ${VERSION} \
    && sh install.sh

ADD Entrypoint.sh /Entrypoint
RUN chmod +x /Entrypoint

HEALTHCHECK CMD netstat -tlnp |grep 8555 || exit 1

EXPOSE 8555 9256

ENTRYPOINT ["/Entrypoint"]

```

 Entrypoint.sh
-----------------------------------------
```
#!/bin/bash

cd ~/chia-blockchain
. ./activate

if [ ! -d ~/.chia/mainnet/config ];then
    chia init
    sed -i 's/localhost/0.0.0.0/g' ~/.chia/mainnet/config/config.yaml
    sed -i 's/127.0.0.1/0.0.0.0/g' ~/.chia/mainnet/config/config.yaml
    #sed -i 's/host: introducer.chia.net/host: introducer-apse.chia.net/g' ~/.chia/mainnet/config/config.yaml
    #chia keys generate
fi

chia start farmer
tail -f /root/.chia/mainnet/log/debug.log
```

log:
------------------------------------------

![image](https://github.com/Chia-Network/chia-blockchain/assets/141826695/59f920fa-e0ad-4884-a134-9308c20077ec)

**Comment by BrandtH22** [comment]
Hey, as a heads up we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia).

As for the issue, are you connecting to peers at all or does the client continually drop them?

It also looks like you built your own docker container for chia, is there a reason you are not using the official docker instance?
https://github.com/Chia-Network/chia-docker

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
